lib/IRGen/GenProto.cpp

```diff
@@ -1792,8 +1792,15 @@ namespace {
       // Add a relative reference to the type, with the type reference
       // kind stored in the flags.
       auto ref = IGM.getTypeEntityReference(
-                   Conformance->getType()->getAnyNominal());
-      B.addRelativeAddress(ref.getValue());
+                   Conformance->getType()->getCanonicalType());
+
+      // Builtin conformance type references are just MetadataKind, which
+      // require no relative reference.
+      if (isa<BuiltinProtocolConformance>(Conformance)) {
+        B.add(ref.getValue());
```

Comment:
It isn't clear what the comment means here. Is `ref.getValue()` the constant `MetadataKind`, or is it still referring to the metadata record?

Reply:
(This could just be residual confusion from me looking at the old `()` specific version of the patch though. In context, it's clear that it's the `MetadataKind`.)

Reply:
My apologizes if the comment wasn't so clear, hopefully if we move the descriptors to the runtime this just falls out.